src/2d/shallow/topo_module.f90

```diff
@@ -401,8 +404,12 @@ end subroutine set_topo_for_dtopo
 
     subroutine read_topo_file(mx,my,topo_type,fname,topo)
 
+#ifdef NETCDF
+        use netcdf
```

Comment:
I agree with you in general (especially when they are used too much) but there is no other way to do this cleanly in Fortran without duplicating a lot of source.  This depends on the pre-processor only a minimal ammount and we should try to keep any use of it that way.